asset-transfer-private-data/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,451 @@
+    const clientOrg1 = await newGrpcConnection(
+        tlsCertPathOrg1,
+        peerEndpointOrg1,
+        peerNameOrg1
+    );
+
+    const gatewayOrg1 = connect({
+        client: clientOrg1,
+        identity: await newIdentity(certPathOrg1, mspIdOrg1),
+        signer: await newSigner(keyDirectoryPathOrg1),
+    });
+
+    // The gRPC client connection from org2 should be shared by all Gateway connections to this endpoint.
+    const clientOrg2 = await newGrpcConnection(
+        tlsCertPathOrg2,
+        peerEndpointOrg2,
+        peerNameOrg2
+    );
+
+    const gatewayOrg2 = connect({
+        client: clientOrg2,
+        identity: await newIdentity(certPathOrg2, mspIdOrg2),
+        signer: await newSigner(keyDirectoryPathOrg2),
+    });
+
+    try {
+        // Get the smart contract from the network.
+        const contractOrg1 = gatewayOrg1.getNetwork(channelName).getContract(chaincodeName);
+
+        // Get the smart contract from the network.
+        const contractOrg2 = gatewayOrg2.getNetwork(channelName).getContract(chaincodeName);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~');
+
+        // Create new assets on the ledger.
+        await createAssets(contractOrg1);
+
+        //Read asset from the org1's private data collection with ID in the given range.
+        await getAssetsByRange(contractOrg1);
+
+        //Attempt to transfer asset without prior aprroval from org2, transaction expected to fail.
+        await transferAssetWithoutAprroval(contractOrg1);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org2 Client ~~~~~~~~~~~~~~~~');
+
+        //Read the asset by ID.
+        await readAssetByID(contractOrg2);
+
+        //Make agreement to transfer the asset from org1 to org2.
+        await agreeToTransfer(contractOrg2);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~');
+
+        //Read transfer agreement.
+        await readTransferAgreement(contractOrg1);
+
+        // Tranfer asset to org2.
+        await transferAsset(contractOrg1);
+
+        //Again ReadAsset : results will show that the buyer identity now owns the asset.
+        await readAssetByID(contractOrg1);
+
+        //Confirm that transfer removed the private details from the Org1 collection.
+        await checkPrivateDataIsRemoved(contractOrg1);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org2 Client ~~~~~~~~~~~~~~~~');
+
+        //Non-owner Org2 should not be able to delete assetID2. Expect an error from DeleteAsset
+        await deleteAssetOrg2(contractOrg2);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~');
+
+        // Delete AssetID2 as Org1.
+        await deleteAssetOrg1(contractOrg1);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org2 Client ~~~~~~~~~~~~~~~~');
+
+        //Org2 can ReadAssetPrivateDetails: Org2 is owner, and private details exist in new owner's Collection
+        await readAssetPrivateDetailsOrg2(contractOrg2);
```

Comment:
It might make the high-level flow clearer here if the function is just named `readAssetPrivateData()` and the collection name (`org2PrivateCollectionName`) is passed as an argument along with the contract.